### PR TITLE
Convert Decimal type, literals, and primitives to Numeric when feature is available.

### PR DESCRIPTION
**Pull request by associahedron**
A better alternative to #2712 for now. This PR updates the LF converter to use the new Numeric type instead of Decimal if it's available, and updates the conversion of Decimal primitives accordingly.

### CODE REVIEW

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Serializability.hs [hunk 1]
```diff
@@ -62,7 +62,7 @@ serializabilityConditionsType world0 _version mbModNameTpls vars = go
           -- requiring 0 <= n <= 'numericMaxScale' for the argument
           -- to Numeric. If the argument isn't given explicitly, we
           -- can't guarantee serializability.
-      TNat _ -> noConditions
+      TNat _ -> Left URTypeLevelNat
```
Comment (associahedron) [inline compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Serializability.hs]: This is an unrelated fix caught by @hurryabit

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs [hunk 2]
```diff
@@ -51,6 +51,12 @@ data Feature = Feature
 -- featureTextCodePoints :: Feature
 -- featureTextCodePoints = Feature "Conversion between text and code points" version1_6
 
+featureNumeric :: Feature
+featureNumeric = Feature
+    { featureName = "Numeric type"
+    , featureMinVersion = versionDev -- TODO: Update when stabilized.
```
Comment (cocreature) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs]: iirc there is a tracking issue for this somewhere right? Maybe link to that?

Comment (hurryabit) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs]: I would add `version1_7 = versionVDev` above, add it to `supportedOutputVersions`, and use `version1_7` here. Then rolling out version 1.7 just requires switching to `version1_7 = V1 $ PointStable 7`.

Comment (associahedron) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs]: Ah, yes! Thanks!

**Comment by remyhaemmerle-da** [comment]
This PR advances the state of #2289.
